src/java.desktop/share/classes/javax/swing/plaf/metal/MetalToolTipUI.java

```diff
@@ -120,6 +118,11 @@ public void paint(Graphics g, JComponent c) {
             insets.top,
             size.width - (insets.left + insets.right) - 6 - accelSpacing,
             size.height - (insets.top + insets.bottom));
+
+        if (paintTextR.width <= 0 || paintTextR.height <= 0) {
+            return;
+        }
```

Comment:
Any idea why only MetalToolTip is affected only? JDK-8040630 fix was generic so shouldn't we apply this fix in BasicToolTipUI#paint instead?

Reply:
Only Metal LaF uses `BasicHTML#getHTMLBaseline` for HTML tooltips painting.

Changing of `BasicToolTipUI#paint` will have no effect since it is not on the call stack.